src/stream/src/executor/actor.rs

```diff
@@ -72,21 +78,33 @@ where
 {
     pub fn new(
         consumer: C,
+        subtasks: Vec<SubtaskHandle>,
         id: ActorId,
         context: Arc<SharedContext>,
         metrics: Arc<StreamingMetrics>,
         actor_context: ActorContextRef,
     ) -> Self {
         Self {
             consumer,
+            subtasks,
             id,
             context,
             _metrics: metrics,
             _actor_context: actor_context,
         }
     }
 
-    pub async fn run(self) -> StreamResult<()> {
+    #[inline(always)]
+    pub async fn run(mut self) -> StreamResult<()> {
+        futures::join!(
+            // Drive the subtasks concurrently.
+            join_all(std::mem::take(&mut self.subtasks)),
```

Comment:
`join_all` will biasedly drive the task at the end of this subtask chain with the highest priority, right?
Guess this is better than the other way around

Reply:
IIUC, https://docs.rs/futures-util/0.2.1/src/futures_util/future/join_all.rs.html#100 will always poll the first one in `self.subtasks` which is the last task in the chain because of the way we push tasks in `create_node_inner`?
Not a big deal

Reply:
As we create nodes recursively, so I think the order of `subtasks` is from the upstream to the downstream? So we'll always poll the most upstream subtask first, which sounds reasonable.